vortex-file/src/v2/file.rs

```diff
@@ -33,6 +34,28 @@ pub struct VortexFile<I> {
     pub(crate) splits: Arc<[Range<u64>]>,
 }
 
+pub struct Scan {
+    projection: ExprRef,
+    filter: Option<ExprRef>,
+}
+
+impl Scan {
+    pub fn all() -> Self {
+        Self {
+            projection: Identity::new_expr(),
+            filter: None,
+        }
+    }
+
+    pub fn new(projection: ExprRef, filter: Option<ExprRef>) -> Self {
+        Self { projection, filter }
+    }
+
+    pub fn build(self, dtype: DType) -> VortexResult<Arc<Scanner>> {
```

Comment:
This shouldn't exist!!

Reply:
create a scanner from a scan and a dtype?